omc3/scripts/linfile_clean.py

```diff
@@ -0,0 +1,209 @@
+            default=0.0,
+        ),
+        backup=dict(
+            action="store_true",
+            help=("Create a backup of the files before overwriting. "
+                  "The backups are numbered, with the highest number being "
+                  "the latest backup.")
+        ),
+    )
+
+
+@entrypoint(get_params(), strict=True)
+def main(opt):
+    """Main function, to parse commandline input and separate restoration
+    from cleaning."""
+    save_config(Path('.'), opt, __file__)
+    if opt.restore:
+        restore_files(opt.files)
+        return
+
+    if opt.columns is None:
+        raise ValueError("The option 'columns' is required for cleaning.")
+    clean_columns(opt.files, opt.columns, opt.limit, opt.backup)
+
+
+# Restore ----------------------------------------------------------------------
+
+def restore_files(files: Sequence[Union[Path, str]]):
+    """Restore backupped files."""
+    failed = []
+    for file in files:
+        file = Path(file)
+        try:
+            _restore_file(file)
+        except IOError as e:
+            failed.append(str(e))
+
+    if len(failed):
+        all_errors = '\n'.join(failed)
+        if len(failed) == len(files):
+            raise IOError(f"Restoration of ALL files has failed\n{all_errors}")
+        raise IOError(f"Restoration of some files has failed, "
+                      f"but the others were restored:\n{all_errors}")
+
+    LOG.info("Restoration successfully completed.")
+
+
+def _restore_file(file):
+    counter = 1
+    backup_file = _get_backup_filepath(file, counter)
+    if not backup_file.exists():
+        raise IOError(f"No backups found for file {file.name}")
+
+    # get last existing backup file
+    while backup_file.exists():
+        counter += 1
+        backup_file = _get_backup_filepath(file, counter)
+    backup_file = _get_backup_filepath(file, counter-1)
+
+    # restore found  file
+    LOG.info(f"Restoring last backup file {backup_file.name} to {file.name}.")
+    shutil.move(backup_file, file)  # overwrites file
+
+
+# Clean ------------------------------------------------------------------------
+
+def clean_columns(files: Sequence[Union[Path, str]], columns: Sequence[str],
+                  limit: float = 0.0, backup: bool = True):
+    """ Clean the columns in the given files."""
+    for file in files:
+        file = Path(file)
+        LOG.info(f"Cleaning {file.name}.")
+        df = tfs.read_tfs(file, index=COL_NAME)
+        for column in columns:
+            df = _filter_by_column(df, column, limit)
+
+        if backup:
+            _backup_file(file)
+
+        tfs.write_tfs(file, df, save_index=COL_NAME)
+
+
+def _filter_by_column(df: pd.DataFrame, column: str, limit: Number) -> pd.DataFrame:
+    """Get the dataframe with all rows dropped filtered by the given column."""
+    if column not in df.columns:
+        LOG.info(f"{column} not in current file. Skipping cleaning.")
```

Comment:
Maybe `warning`?

Reply:
No. Probably on purpose, if you want to clean `linx` and `liny` files in one go (I will call it twice from the GUI though). Then you can give it "TUNEX" and "TUNEY" as columns and it will just skipt it